core/arch/arm/plat-imx6/imx6_init.S

```diff
@@ -0,0 +1,346 @@
+	 *
+	 * PCR = 0x00000001
+	 * - no change latency, enable clk gating
+	 */
+	movw r0, #0x4000
+	movt r0, #0x0000
+	write_sctlr r0
+
+	movw r0, #0x0041
+	movt r0, #0x0000
+	write_actlr r0
+
+	movw r0, #0x0C00
+	movt r0, #0x0002
+	write_nsacr r0
+
+	movw r0, #0x0000
+	movt r0, #0x0001
+	write_pcr r0
+
+	/*
+	 * GIC configuration
+	 *
+	 * Register ICDISR0 = 0xFFFFFFFF
+	 * - All local interrupts are NonSecure.
+	 *
+	 * Register ICCPMR = 0xFFFFFFFF
+	 */
+
+	ldr  r0, =GIC_DIST_BASE
+	mov  r1, #0xFFFFFFFF
+	str  r1, [r0, #GIC_DIST_ISR0]
+
+	ldr  r0, =GIC_CPU_BASE
+	mov  r1, #0xFFFFFFFF
+	str  r1, [r0, #CORE_ICC_ICCPMR]
+
+	mov pc, lr /* back to tzinit */
+UNWIND(	.fnend)
+END_FUNC plat_cpu_reset_early
+
+/*
+ * A9 secured config, needed only from a single core
+ *
+ * Use scratables registers R0-R3.
+ * No stack usage.
+ * LR store return address.
+ * Trap CPU in case of error.
+ *
+ * TODO: size optim in code
+ */
+FUNC plat_cpu_reset_late , :
+UNWIND(	.fnstart)
+
+	mrc p15, 0, r0, c0, c0, 5
+	ands r0, #3
+	beq _boot_late_primary_cpu
+
+_boot_late_secondary_cpu:
+	mov pc, lr
+
+_boot_late_primary_cpu:
+
+	/*
+	 * Snoop Control Unit configuration
+	 *
+	 * SCU is enabled with filtering off.
+	 * Both Secure/Unsecure can access SCU and timers
+	 *
+	 * 0x00 SCUControl = 0x00000060 !!!  should be 0x5 !
+	 * 0x04 SCUConfiguration =  ???
+	 * 0x0C SCUInvalidateAll (Secure cfg)
+	 * 0x40 FilteringStartAddress = 0x40000000
+	 * 0x44 FilteeringEndAddress - 0x80000000
+	 * 0x50 SCUAccessControl
+	 * 0x54 SCUSecureAccessControl
+	 */
+
+	/*
+	 * SCU Access Register : SAC = 0x0000000F
+	 * - both secure CPU access SCU
+	 */
+	ldr  r0, =SCU_BASE
+	movw r1, #0x000F
+	movt r1, #0x0000
+	str  r1, [r0, #SCU_SAC]
+
+	/*
+	 * SCU NonSecure Access Register : SNSAC : 0x00000FFF
+	 * - both nonsec cpu access SCU, private and global timer
+	 */
+	movw r1, #0x0FFF
+	movt r1, #0x0000
+	str  r1, [r0, #SCU_NSAC]
+
+	/*- GIC secure configuration ---*/
+
+	/*
+	 * Register ICDISR[1-31] = 0xFFFFFFFF
+	 * - All external interrupts are NonSecure.
+	 */
+	ldr  r0, =(GIC_DIST_BASE + GIC_DIST_ISR1)
+	mov  r2, #0xFFFFFFFF
+	mov  r1, #31 /* Nb of loop rounds */
+loop_1:
+	str r2, [r0]
+	add r0, #4
+	sub r1, r1, #1
+	cmp r1, #0
+	bne loop_1
+
+	/* Allow NSec to manage FIQ/Imprecise abort */
+	mrc p15, 0, r0, c1, c1, 0    /* read Secure Configuration Register */
+	orr r0, r0, #0x30            /* SCR[FW]=1, SCR[AW]=1 */
+	mcr p15, 0, r0, c1, c1, 0    /* write Secure Configuration Register */
+
+    /* configure imx6 CSU */
+
+    /* first grant access of all peripherals to NS... */
+    ldr r0, =0x021c0000
+    ldr r1, =0x021c00a0
+    ldr r2, =0xffffffff
+
+loop_csu:
+    str r2, [r0]
+    add r0, r0, #4
+    cmp r0, r1
+    bne loop_csu
+
+    /* then restrict key peripherals */
+
+    ldr r0, =0x021c0014 /* POWER group */
+    ldr r2, =0x003300ff
+    str r2, [r0]
+
+    ldr r0, =0x021c0040 /* TZASC1,TZASC2 */
+    ldr r2, =0x00330033
+    str r2, [r0]
+
+    /* TODO: TZASC init */
```

Comment:
No, this is actually a real TODO thing. We are going to configure TZASC to divide the DDR into NS DDR and Sec DDR, which agrees with the DDR space definition in plat-xxx/platfom_config.h. For now the DDR separation is not protected by TZASC.